database/src/main/java/com/firebase/ui/database/FirebaseIndexArray.java

```diff
@@ -97,7 +100,10 @@ public void onChildChanged(EventType type, DataSnapshot snapshot, int index, int
 
     @Override
     public void onDataChanged() {
-        // No-op, we don't listen to batch events for the key ref
+        if (mHasQueuedInstantOps) {
+            notifyListenersOnDataChanged();
```

Comment:
@samtstern No, there shouldn't be. For deletions, there will be two notifications if the dev deletes the content and then the key. However, if they do it the suggested way of deleting the key first and then the data, they'll be fine. As for moves, there definitely will only be one notification.